pkg/apis/triggers/config/feature_flags.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	StableAPIFields        = "stable"
+	AlphaAPIFields         = "alpha"
+	enableAPIFields        = "enable-api-fields"
+	DefaultEnableAPIFields = StableAPIFields
+)
+
+// FeatureFlags holds the features configurations
+// +k8s:deepcopy-gen=true
+type FeatureFlags struct {
+	EnableAPIFields string
+}
+
+// GetFeatureFlagsConfigName returns the name of the configmap containing all
+// feature flags.
+func GetFeatureFlagsConfigName() string {
+	if e := os.Getenv("CONFIG_FEATURE_FLAGS_NAME"); e != "" {
+		return e
+	}
+	return "feature-flags-triggers"
+}
+
+// NewFeatureFlagsFromMap returns a Config given a map corresponding to a ConfigMap
+func NewFeatureFlagsFromMap(cfgMap map[string]string) (*FeatureFlags, error) {
+	tc := FeatureFlags{}
+	if err := setEnabledAPIFields(cfgMap, DefaultEnableAPIFields, &tc.EnableAPIFields); err != nil {
```

Comment:
I'm having trouble seeing the value of using a field ref here over just setting the field directly. Can you add some detail about why it is structured this way, or is this equivalent?

```go
var err error
tc.EnableAPIFields, err = getEnabledAPI(cfgMap)
```

Also removed the default param - setEnabledAPIFields is already specific to the flag because it needs to specify the `enableAPIFields` key - we can do something similar for the default value as well.

Reply:
done.